build.gradle

```diff
@@ -494,18 +537,19 @@ project(':tools') {
         include "**/org/apache/kafka/tools/*"
     }
 
-    tasks.create(name: "copyDependantLibs", type: Copy) {
-        from (configurations.testRuntime) {
-            include('slf4j-log4j12*')
-        }
-        from (configurations.runtime) {
-            exclude('kafka-clients*')
-        }
-        into "$buildDir/dependant-libs-${versions.scala}"
-    }
+    task linkJars(dependsOn: ['copyDependantLibsJava', 'copyTestDependantLibsJava']) {
+      Files.createDirectories(Paths.get("$buildDir/dependant-libs-java"))
```

Comment:
All of the work inside the linkJars task should be wrapped in `doLast { ... }` to prevent it from from executing at configuration time. 

You can see this break by running `gradle clean build -x test`. It will show a FileNotFound exception because at configuration time the copyDependantLibs tasks haven't run yet. Make sure you clean each time when testing changes to catch things like this.